server/etcdserver/cindex/cindex.go

```diff
@@ -58,6 +64,19 @@ type consistentIndex struct {
 	// The value is being persisted in the backend since v3.5.
 	term uint64
 
+	// applyingIndex and applyingTerm are just temporary cache of the raftpb.Entry.Index
+	// and raftpb.Entry.Term, and they are not ready to be persisted yet. They will be
+	// saved to consistentIndex and term above in the txPostLockInsideApplyHook.
+	//
+	// TODO(ahrtr): try to remove the OnPreCommitUnsafe, and compare the
```

Comment:
Can we do the performance check before merging this PR? I would prefer not to backport a PR with todo. Please let me know if you need help with measuring it.

Reply:
I think we are all good so far. I will investigate whether or not we should completely remove the `OnPreCommitUnsafe` in the next step **only** in the **main** branch.  The change will not be cherry picked to 3.5. 

We can use the tools/benchmark, correct?

Reply:
I wanted to backport this change because it makes the code much simpler, which will be important for long term maintenance of release-3.5 branch. Just making sure that the decision not to fix it imminently is not rushed, let's not sacrifice the quality, however it you think it's not worth backporting that's also ok.

Reply:
Your comment is valid, but usually we only backport bug fixes to previous release, so does this PR. But completely possibly removing `OnPreCommitUnsafe` is a refactor, and so I don't think we should backport it to 3.5.  Please note that refactoring may also introduce regression, so we should only do it in main branch.

Reply:
I think this PR is good to go.